File: 137_single_number_2/singleNumberMap.ts
function singleNumber(nums: number[]): number {
  const map = {};

  for (let num of nums) {
    const value = map[num];
    if (value) {
      map[num] = value + 1;
    } else {
      map[num] = 1;
    }
  }

  for (const [key, value] of Object.entries(map)) {
    if (value == 1) {
      return +key;
    }
  }

  return 0;
}

File: 137_single_number_2/singleNumberMap.py
from typing import List


class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        map = {}

        for num in nums:
            map[num] = map.get(num, 0) + 1

        for key, value in map.items():
            if value == 1:
                return key